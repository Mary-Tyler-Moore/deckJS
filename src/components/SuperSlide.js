import React from 'react'
import { SlideTitle } from './'
import { faChevronRight, faChevronLeft } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Container, Columns, Column, Hero, HeroBody } from 'bloomer'

class SuperSlide extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      location: 0
    };
  }

  nextSlide() {
    this.setState({
      location: (this.state.location + 1)
    })
  }

  prevSlide() {
    this.setState({
      location: (this.state.location - 1)
    })
  }

  arrowsMagic() {
    if (this.state.location == 0) {
      return (
        <Columns isCentered>
          <Column className='is-one-fifth'>
          </Column>
          <Column style={{display: 'flex', justifyContent: 'center'}}>
            {this.props.content[0]()}
          </Column>
          <Column style={{display: 'flex', alignItems: 'center', justifyContent: 'flex-end'}} className='is-one-fifth'>
            {
              (this.state.location == (this.props.content.length - 1)) ?
              '' :
              <a onClick={this.nextSlide.bind(this)} id='slide-next' >
                <FontAwesomeIcon
                  className="has-text-light"
                  icon={faChevronRight}
                  size="3x"
                  fixedWidth
                />
              </a>
            }
          </Column>
        </Columns>
      )
    } else {
      return (
        <Columns isCentered>
          <Column style={{display: 'flex', alignItems: 'center'}} className='is-one-fifth'>
            <a onClick={this.prevSlide.bind(this)} id='slide-prev' >
              <FontAwesomeIcon
                className="has-text-light"
                icon={faChevronLeft}
                size="3x"
                fixedWidth
              />
            </a>
          </Column>
          <Column style={{display: 'flex', justifyContent: 'center'}}>
            {this.props.content[this.state.location]()}
          </Column>
          <Column style={{display: 'flex', alignItems: 'center', justifyContent: 'flex-end'}} className='is-one-fifth'>
            {
              (this.state.location == (this.props.content.length - 1)) ?
              '' :
              <a onClick={this.nextSlide.bind(this)} id='slide-next' >
                <FontAwesomeIcon
                  className="has-text-light"
                  icon={faChevronRight}
                  size="3x"
                  fixedWidth
                />
              </a>
            }
          </Column>
        </Columns>
      )
    }
  }

  render() {
    return (
      <Hero isFullHeight isColor={this.props.color} key={this.props.key}>
        <HeroBody style={{ display: 'flex', flexDirection: 'column' }}>
          <SlideTitle title={this.props.title} subtitle={this.props.subtitle} />
          <Container hasTextAlign="centered">
            {this.arrowsMagic()}
          </Container>
        </HeroBody>
      </Hero>
    )
  }

}

export default SuperSlide
