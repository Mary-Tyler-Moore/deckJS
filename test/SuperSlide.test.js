import React from 'react'
import SuperSlide from '../src/components/SuperSlide'
import { shallow, mount } from 'enzyme'

describe('SuperSlide', () => {
  const mock = {
    title: 'Test',
    subtitle: 'Substitle',
    color: 'color',
    content: [() => {}, () => {}]
  }

  it('should render', () => {
    const wrapper = shallow(<SuperSlide {...mock} />)
    expect(wrapper).toMatchSnapshot()
  })
})