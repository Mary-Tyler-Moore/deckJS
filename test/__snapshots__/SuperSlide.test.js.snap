// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuperSlide should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SuperSlide
    color="color"
    content={
      Array [
        [Function],
        [Function],
      ]
    }
    subtitle="Substitle"
    title="Test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "Test",
    "nodeType": "function",
    "props": Object {
      "children": <withHelpersModifiers(HeroBody)
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <SlideTitle
          subtitle="Substitle"
          title="Test"
        />
        <withHelpersModifiers(Container)
          hasTextAlign="centered"
        >
          <withHelpersModifiers(Columns)
            isCentered={true}
          >
            <withHelpersModifiers(Column)
              className="is-one-fifth"
            />
            <withHelpersModifiers(Column)
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            />
            <withHelpersModifiers(Column)
              className="is-one-fifth"
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "flex-end",
                }
              }
            >
              <a
                onClick={[Function]}
              >
                <FontAwesomeIcon
                  border={false}
                  className="has-text-light"
                  fixedWidth={true}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        320,
                        512,
                        Array [],
                        "f054",
                        "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                      ],
                      "iconName": "chevron-right",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="3x"
                  spin={false}
                  symbol={false}
                  transform={null}
                />
              </a>
            </withHelpersModifiers(Column)>
          </withHelpersModifiers(Columns)>
        </withHelpersModifiers(Container)>
      </withHelpersModifiers(HeroBody)>,
      "isColor": "color",
      "isFullHeight": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <SlideTitle
            subtitle="Substitle"
            title="Test"
          />,
          <withHelpersModifiers(Container)
            hasTextAlign="centered"
          >
            <withHelpersModifiers(Columns)
              isCentered={true}
            >
              <withHelpersModifiers(Column)
                className="is-one-fifth"
              />
              <withHelpersModifiers(Column)
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              />
              <withHelpersModifiers(Column)
                className="is-one-fifth"
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <a
                  onClick={[Function]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className="has-text-light"
                    fixedWidth={true}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          320,
                          512,
                          Array [],
                          "f054",
                          "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                        ],
                        "iconName": "chevron-right",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size="3x"
                    spin={false}
                    symbol={false}
                    transform={null}
                  />
                </a>
              </withHelpersModifiers(Column)>
            </withHelpersModifiers(Columns)>
          </withHelpersModifiers(Container)>,
        ],
        "style": Object {
          "display": "flex",
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "subtitle": "Substitle",
            "title": "Test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <withHelpersModifiers(Columns)
              isCentered={true}
            >
              <withHelpersModifiers(Column)
                className="is-one-fifth"
              />
              <withHelpersModifiers(Column)
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              />
              <withHelpersModifiers(Column)
                className="is-one-fifth"
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <a
                  onClick={[Function]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className="has-text-light"
                    fixedWidth={true}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          320,
                          512,
                          Array [],
                          "f054",
                          "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                        ],
                        "iconName": "chevron-right",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size="3x"
                    spin={false}
                    symbol={false}
                    transform={null}
                  />
                </a>
              </withHelpersModifiers(Column)>
            </withHelpersModifiers(Columns)>,
            "hasTextAlign": "centered",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <withHelpersModifiers(Column)
                  className="is-one-fifth"
                />,
                <withHelpersModifiers(Column)
                  style={
                    Object {
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                />,
                <withHelpersModifiers(Column)
                  className="is-one-fifth"
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "justifyContent": "flex-end",
                    }
                  }
                >
                  <a
                    onClick={[Function]}
                  >
                    <FontAwesomeIcon
                      border={false}
                      className="has-text-light"
                      fixedWidth={true}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            320,
                            512,
                            Array [],
                            "f054",
                            "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                          ],
                          "iconName": "chevron-right",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size="3x"
                      spin={false}
                      symbol={false}
                      transform={null}
                    />
                  </a>
                </withHelpersModifiers(Column)>,
              ],
              "isCentered": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "is-one-fifth",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": undefined,
                  "style": Object {
                    "display": "flex",
                    "justifyContent": "center",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <a
                    onClick={[Function]}
                  >
                    <FontAwesomeIcon
                      border={false}
                      className="has-text-light"
                      fixedWidth={true}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            320,
                            512,
                            Array [],
                            "f054",
                            "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                          ],
                          "iconName": "chevron-right",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size="3x"
                      spin={false}
                      symbol={false}
                      transform={null}
                    />
                  </a>,
                  "className": "is-one-fifth",
                  "style": Object {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "flex-end",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <FontAwesomeIcon
                      border={false}
                      className="has-text-light"
                      fixedWidth={true}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            320,
                            512,
                            Array [],
                            "f054",
                            "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                          ],
                          "iconName": "chevron-right",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size="3x"
                      spin={false}
                      symbol={false}
                      transform={null}
                    />,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "border": false,
                      "className": "has-text-light",
                      "fixedWidth": true,
                      "flip": null,
                      "icon": Object {
                        "icon": Array [
                          320,
                          512,
                          Array [],
                          "f054",
                          "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                        ],
                        "iconName": "chevron-right",
                        "prefix": "fas",
                      },
                      "inverse": false,
                      "listItem": false,
                      "mask": null,
                      "pull": null,
                      "pulse": false,
                      "rotation": null,
                      "size": "3x",
                      "spin": false,
                      "symbol": false,
                      "transform": null,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "a",
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "Test",
      "nodeType": "function",
      "props": Object {
        "children": <withHelpersModifiers(HeroBody)
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <SlideTitle
            subtitle="Substitle"
            title="Test"
          />
          <withHelpersModifiers(Container)
            hasTextAlign="centered"
          >
            <withHelpersModifiers(Columns)
              isCentered={true}
            >
              <withHelpersModifiers(Column)
                className="is-one-fifth"
              />
              <withHelpersModifiers(Column)
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              />
              <withHelpersModifiers(Column)
                className="is-one-fifth"
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <a
                  onClick={[Function]}
                >
                  <FontAwesomeIcon
                    border={false}
                    className="has-text-light"
                    fixedWidth={true}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          320,
                          512,
                          Array [],
                          "f054",
                          "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                        ],
                        "iconName": "chevron-right",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size="3x"
                    spin={false}
                    symbol={false}
                    transform={null}
                  />
                </a>
              </withHelpersModifiers(Column)>
            </withHelpersModifiers(Columns)>
          </withHelpersModifiers(Container)>
        </withHelpersModifiers(HeroBody)>,
        "isColor": "color",
        "isFullHeight": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <SlideTitle
              subtitle="Substitle"
              title="Test"
            />,
            <withHelpersModifiers(Container)
              hasTextAlign="centered"
            >
              <withHelpersModifiers(Columns)
                isCentered={true}
              >
                <withHelpersModifiers(Column)
                  className="is-one-fifth"
                />
                <withHelpersModifiers(Column)
                  style={
                    Object {
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                />
                <withHelpersModifiers(Column)
                  className="is-one-fifth"
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "justifyContent": "flex-end",
                    }
                  }
                >
                  <a
                    onClick={[Function]}
                  >
                    <FontAwesomeIcon
                      border={false}
                      className="has-text-light"
                      fixedWidth={true}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            320,
                            512,
                            Array [],
                            "f054",
                            "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                          ],
                          "iconName": "chevron-right",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size="3x"
                      spin={false}
                      symbol={false}
                      transform={null}
                    />
                  </a>
                </withHelpersModifiers(Column)>
              </withHelpersModifiers(Columns)>
            </withHelpersModifiers(Container)>,
          ],
          "style": Object {
            "display": "flex",
            "flexDirection": "column",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "subtitle": "Substitle",
              "title": "Test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <withHelpersModifiers(Columns)
                isCentered={true}
              >
                <withHelpersModifiers(Column)
                  className="is-one-fifth"
                />
                <withHelpersModifiers(Column)
                  style={
                    Object {
                      "display": "flex",
                      "justifyContent": "center",
                    }
                  }
                />
                <withHelpersModifiers(Column)
                  className="is-one-fifth"
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "justifyContent": "flex-end",
                    }
                  }
                >
                  <a
                    onClick={[Function]}
                  >
                    <FontAwesomeIcon
                      border={false}
                      className="has-text-light"
                      fixedWidth={true}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            320,
                            512,
                            Array [],
                            "f054",
                            "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                          ],
                          "iconName": "chevron-right",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size="3x"
                      spin={false}
                      symbol={false}
                      transform={null}
                    />
                  </a>
                </withHelpersModifiers(Column)>
              </withHelpersModifiers(Columns)>,
              "hasTextAlign": "centered",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <withHelpersModifiers(Column)
                    className="is-one-fifth"
                  />,
                  <withHelpersModifiers(Column)
                    style={
                      Object {
                        "display": "flex",
                        "justifyContent": "center",
                      }
                    }
                  />,
                  <withHelpersModifiers(Column)
                    className="is-one-fifth"
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "justifyContent": "flex-end",
                      }
                    }
                  >
                    <a
                      onClick={[Function]}
                    >
                      <FontAwesomeIcon
                        border={false}
                        className="has-text-light"
                        fixedWidth={true}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              320,
                              512,
                              Array [],
                              "f054",
                              "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                            ],
                            "iconName": "chevron-right",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size="3x"
                        spin={false}
                        symbol={false}
                        transform={null}
                      />
                    </a>
                  </withHelpersModifiers(Column)>,
                ],
                "isCentered": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "is-one-fifth",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": undefined,
                    "style": Object {
                      "display": "flex",
                      "justifyContent": "center",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <a
                      onClick={[Function]}
                    >
                      <FontAwesomeIcon
                        border={false}
                        className="has-text-light"
                        fixedWidth={true}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              320,
                              512,
                              Array [],
                              "f054",
                              "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                            ],
                            "iconName": "chevron-right",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size="3x"
                        spin={false}
                        symbol={false}
                        transform={null}
                      />
                    </a>,
                    "className": "is-one-fifth",
                    "style": Object {
                      "alignItems": "center",
                      "display": "flex",
                      "justifyContent": "flex-end",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <FontAwesomeIcon
                        border={false}
                        className="has-text-light"
                        fixedWidth={true}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              320,
                              512,
                              Array [],
                              "f054",
                              "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                            ],
                            "iconName": "chevron-right",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size="3x"
                        spin={false}
                        symbol={false}
                        transform={null}
                      />,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "border": false,
                        "className": "has-text-light",
                        "fixedWidth": true,
                        "flip": null,
                        "icon": Object {
                          "icon": Array [
                            320,
                            512,
                            Array [],
                            "f054",
                            "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                          ],
                          "iconName": "chevron-right",
                          "prefix": "fas",
                        },
                        "inverse": false,
                        "listItem": false,
                        "mask": null,
                        "pull": null,
                        "pulse": false,
                        "rotation": null,
                        "size": "3x",
                        "spin": false,
                        "symbol": false,
                        "transform": null,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "a",
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
